JCDispatch/src/main/java/lib/nameParsing.java

```diff
@@ -0,0 +1,155 @@
+                return fNames[ 0 ].trim();
+            }
+
+            String[] fullNames = strNames[ 0 ].split( " " );
+            return fullNames[ 0 ].trim();
+        }
+        return null;
+    }
+
+    public static String getLastName( Object obj ) {
+        if ( obj != null ) {
+            String name = formatText( obj );
+            name = removeDigit( name );
+            Matcher m = p.matcher( name );
+
+            if ( m.find() ) {
+                return "";
+            }
+
+            String[] strNames = name.split( "," );
+            if ( strNames.length > 1 ) {
+                return strNames[ 0 ].trim();
+            }
+            String[] fullNames = strNames[ 0 ].split( " " );
+            return fullNames[ fullNames.length - 1 ].trim();
+        }
+        return null;
+    }
+
+    public static String getMiddleName( Object obj ) {
+        if ( obj != null ) {
+            String name = formatText( obj );
+            name = removeDigit( name );
+            Matcher m = p.matcher( name );
+
+            if ( m.find() ) {
+                return "";
+            }
+
+            String[] strNames = name.split( "," );
+            if ( strNames.length > 1 ) {
+                String fName = strNames[ 1 ].trim();
+                List<String> fNames = Arrays.asList( fName.split( " " ) );
+                if ( fNames.size() > 2 ) {
+                    List<String> mNames = fNames.subList( 1, fNames.size() );
+                    return String.join( " ", mNames ).trim();
+                }
+                if ( fNames.size() == 2 ) {
+                    return fNames.get( fNames.size() - 1 ).trim();
+                }
+                return "";
+            }
+            String[] middleNames = strNames[ 0 ].split( " " );
+            if ( middleNames.length > 2 ) {
+                return middleNames[ 1 ].trim();
+            }
+        }
+        return null;
+    }
+
+    public static String getName( Object obj ) {
+        if ( obj != null ) {
+            String name = obj.toString().trim();
+            Matcher m = p.matcher( name );
+
+            if ( m.find() ) {
+                return name;
+            }
+            return "";
+        }
+        return null;
+    }
+
+    public static String getAddressID( Object obj ) {
+        if ( obj != null ) {
+            String address = obj.toString().trim();
+            if ( address.contains( "Ia" ) || address.contains( "null" ) ) {
+                address = address.replace( "Ia", "IA" );
+                address = address.replace( "null", "" );
+                return String.join( "" , Arrays.asList( address.split( " " ) ) );
+            }
+            return String.join( "" , Arrays.asList( address.split( " " ) ) );
+        }
+        return null;
+    }
+
+    public static String getIntersection( Object obj ) {
+        if ( obj != null ) {
+            String address = obj.toString().trim();
+            if ( address.contains( "/" ) ) {
```

Comment:
This if statement is redundant. Contains will perform a similar linear search and will do nothing if character is not found so test is unnecessary.

Reply:
I want the values that contain "/" otherwise return "". How do I do that without an if statement?